gpt_index/indices/node_utils.py

```diff
@@ -6,31 +6,58 @@
 
 from gpt_index.data_structs.data_structs import Node
 from gpt_index.indices.utils import truncate_text
-from gpt_index.langchain_helpers.text_splitter import TokenTextSplitter
+from gpt_index.langchain_helpers.text_splitter import TextSplit, TextSplitter
 from gpt_index.schema import BaseDocument
 
 
+def get_text_splits_from_document(
+    document: BaseDocument,
+    text_splitter: TextSplitter,
+    include_extra_info: bool = True,
+) -> List[TextSplit]:
+    """Break the document into chunks with additional info."""
+    # TODO: clean up since this only exists due to the diff w LangChain's TextSplitter
```

Comment:
lol yeah this logic is really hacky. our existing logic about calculating num char overlap in TextSplit breaks with other text splitters. i'll figure out a way to clean this up later